plugins/thunderbolt/fu-thunderbolt-device.c

```diff
@@ -319,6 +319,133 @@ fu_thunderbolt_device_get_attr_uint16(FuThunderboltDevice *self, const gchar *na
 	}
 	return (guint16)val;
 }
+static gboolean
+fu_thunderbolt_check_usb4_port_path(FuUdevDevice *device, const gchar *attribute, GError **err)
+{
+	g_autofree const gchar *path =
+	    g_build_filename(fu_udev_device_get_sysfs_path(device), attribute, NULL);
+	g_autofree gchar *fn = g_strdup_printf("%s", path);
+	g_autoptr(GFile) file = g_file_new_for_path(fn);
+	if (!g_file_query_exists(file, NULL)) {
+		g_set_error(err, G_IO_ERROR, G_IO_ERROR_NOT_FOUND, "failed to find %s", fn);
+		return FALSE;
+	}
+	return TRUE;
+}
+
+static gboolean
+fu_thunderbolt_device_set_port_offline(FuUdevDevice *device, GError **err)
+{
+	const gchar *offline = "usb4_port1/offline";
+	const gchar *rescan = "usb4_port1/rescan";
+	if (fu_thunderbolt_check_usb4_port_path(device, offline, err)) {
```

Comment:
This particular one I wouldn't bother having an error argument or you need to pass a local error since you're going to return true if the file doesn't exist.

Reply:
fu_thunderbolt_check_usb4_port_path has err argument already !

Reply:
You need to pass a local error so you don't set error on true condition

Reply:
got it ok as done in https://github.com/fwupd/fwupd/commit/d4cca8d8eb125b4e86abc8b7cff2e7192bb664d3 ?